### PR TITLE
Devops 272 public instances

### PR DESCRIPTION
**Pull request by Chisomchima**
Implements [DEVOPS-272](https://dhis2.atlassian.net/browse/DEVOPS-272)
---

### Description
Similar to [this](https://play.dhis2.org/) this PR creates a page within the instance manager for listing of public instances. 

---

### Key Improvements

-   [x] Create a public Layout that will house Public Pages.
-   [x] instance table api fetching
-   [x] Display of filtered data on the public instance page.
-   [x] UX upgrade from previous use case. 

---

### Screenshots

Uploading Screen Recording 2024-09-13 at 09.49.54.mov…

### CODE REVIEW

File: src/app/app.tsx [hunk 1]
```diff
@@ -33,6 +33,9 @@ const router = createBrowserRouter(
             <Route path="/reset-password/:token" element={<ResetPassword />} />
             <Route path="/validate/success" element={<ValidateSuccess />} />
             <Route path="/validate/:token" element={<Validate />} />
+            <Route path="/public" element={<PublicLayout />}>
```
Comment (tonsV2) [inline src/app/app.tsx:36]: Since this route is somewhat unrelated to the rest of the routes here I suggest moving it to the top (line 31).

Where has the route for /sign-in gone?

**Review by tonsV2 — CHANGES_REQUESTED** [review]

File: src/components/login.module.css [hunk 2]
```diff
@@ -2,7 +2,7 @@
     display: flex;
     align-items: center;
     justify-content: center;
-    height: 100vh;
+    height: calc(100vh - 65px);
```
Comment (Mohammer5) [inline src/components/login.module.css:5]: Where does the value `65` come from?

File: src/components/public-layout.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    if (location.pathname === '/') {
+        return <Navigate to="/sign-in" />
+    }
+
+    return (
+        <div className={styles.container}>
+            <div className={styles.nav}>
+                <h1 className={styles.header}>
+                    <LogoIcon className={styles.logo} />
+                    DHIS2 Playground
```
Comment (Mohammer5) [inline src/components/public-layout.tsx:18]: Should this be "DHIS2 Playground"? (cc @tonsV2).
There are some orgs out there that use DHIS2 but have changed their logo, should this be done here too (incl. the displayed text?)?

File: src/views/public-instances/list/instance-table-filters.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export interface GroupWithCategories {
+    name: string
+    description: string
+    categories: Category[]
+}
+
+// Hook for fetching instance table data
+export const useInstanceTableData = () => {
```
Comment (Mohammer5) [inline src/views/public-instances/list/instance-table-filters.tsx:21]: File has a different name than the exported function/hooks

File: src/views/reset-password/request-password-reset.module.css [hunk 5]
```diff
@@ -2,7 +2,7 @@
     display: flex;
     align-items: center;
     justify-content: center;
-    height: 100vh;
+    height: calc(100vh - 65px);
```
Comment (Mohammer5) [inline src/views/reset-password/request-password-reset.module.css:5]: Same question as above

File: src/views/reset-password/reset-password.module.css [hunk 6]
```diff
@@ -2,7 +2,7 @@
     display: flex;
     align-items: center;
     justify-content: center;
-    height: 100vh;
+    height: calc(100vh - 65px);
```
Comment (Mohammer5) [inline src/views/reset-password/reset-password.module.css:5]: See above

File: src/views/sign-up/sign-up.module.css [hunk 7]
```diff
@@ -2,7 +2,7 @@
     display: flex;
     align-items: center;
     justify-content: center;
-    height: 100vh;
+    height: calc(100vh - 65px);
```
Comment (Mohammer5) [inline src/views/sign-up/sign-up.module.css:5]: See above

File: src/components/public-layout.tsx (hunk 3 above)
Comment (Mohammer5) [reply src/components/public-layout.tsx:18]: Will be done later, recorded in https://dhis2.atlassian.net/browse/DEVOPS-441

**Review by Mohammer5 — APPROVED** [review]

**Review by tonsV2 — APPROVED** [review]
